### PR TITLE
feat: redirect to first conversation in list

### PR DESCRIPTION
**Pull request by neekolas**
## What does this PR do?

- Attempts to redirect to the first conversation in the list
- Conditions to allow a redirect are
1. The user's viewport is 1024px or greater (we don't want this behaviour on mobile, since we use the `/messages` as a distinct route to show the conversation list)
2. The user is on the `/messages` route
3. There is at least one conversation in the list loaded.

I think this feature is going to get picked up in a planned refactor to merge the `Message.tsx` and `index.tsx` pages into a single route. Right now, the necessary information to execute this redirect is lower in the React tree than is ideal. 

I also had to add a new React hook to track the window width to avoid this behaviour on mobile, which is unfortunate. Open to other solutions that don't rely on `window.innerWidth`.

The PR also changes the send button to only icon in the mobile view and makes the mobile view flow start from 1024px

Fixes # (issue)

https://user-images.githubusercontent.com/65710/199074880-b8df72ff-104b-45ec-9e87-af2e64cf9976.mp4

image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- [ ] Go to the `/messages` page on the desktop. The first conversation should be automatically selected once loading is complete
- [ ] Go to the `/messages` page on mobile. The conversation list should appear, and it should be possible to navigate to the individual conversation and back.
- [ ] In mobile view the send button should only have an icon

### CODE REVIEW

File: src/components/Messages/PreviewList.tsx [hunk 1]
```diff
@@ -7,19 +7,22 @@ import { GridItemFour } from '@components/UI/GridLayout';
...
 import { useRouter } from 'next/router';
 import type { FC } from 'react';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import Custom404 from 'src/pages/404';
 import Custom500 from 'src/pages/500';
 import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 
+const MIN_WIDTH_FOR_TWO_PANELS = 1024;
```
Comment (bhavya2611) [inline src/components/Messages/PreviewList.tsx:24]: I think the mobile flow starts 768px, we should have that here

**Review by bhavya2611 — APPROVED** [review]
Small comment

File: src/components/Messages/PreviewList.tsx (hunk 1 above)
Comment (neekolas) [reply src/components/Messages/PreviewList.tsx:24]: So, maybe it should start at 768. RIght now, between 768 and 1024 it doesn't hide the conversation list when you have a conversation loaded. I did make a separate ticket for that. https://github.com/xmtp-labs/hq/issues/820

Comment (bhavya2611) [reply src/components/Messages/PreviewList.tsx:24]: Also move the constant to the constants file

Comment (neekolas) [reply src/components/Messages/PreviewList.tsx:24]: Ya, that's a good call

Comment (bhavya2611) [reply src/components/Messages/PreviewList.tsx:24]: Okay so are you suggesting we change to a mobile view below 1024px?

I think that will make sense.

**Comment by bigint** [comment]
Can you please resolve the conflicts 🙏🏼

**Comment by bhavya2611** [comment]
@bigint resolved conflicts

File: src/components/utils/hooks/useWindowSize.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+
+interface Size {
+  width: number | undefined;
+  height: number | undefined;
+}
+
+// Borrowed from https://usehooks.com/useWindowSize/
+const useWindowSize = (): Size => {
```
[the remark this answers is not in the document]
Comment (neekolas) [reply src/components/utils/hooks/useWindowSize.tsx:9]: For styling, absolutely. But this is used to change behavior

**Pull request merged**